src/microbe_stage/systems/FluidCurrentsSystem.cs

```diff
@@ -116,7 +180,19 @@ protected override void Update(float delta, in Entity entity)
         var pos = new Vector2(position.Position.X, position.Position.Z);
         var vel = VelocityAt(pos) * Constants.MAX_FORCE_APPLIED_BY_CURRENTS;
 
-        physicsControl.ImpulseToGive += new Vector3(vel.X, 0, vel.Y) * delta;
+        physicsControl.ImpulseToGive += new Vector3(vel.X, 0, vel.Y) * delta
+            * entity.Get<CurrentAffected>().EffectStrength;
         physicsControl.PhysicsApplied = false;
     }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private float GetPixel(float x, float y, float z, Image[] array)
+    {
+        if (x < 0.0f)
+            x = noiseWidth + x % noiseWidth;
```

Comment:
Doesn't this run into negative value modulo differences if x is somehow less than `-noiseWidth`?